.github/workflows/php-test.yml

```diff
@@ -1,145 +1,21 @@
-                restore-keys: |
-                            php${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-
-
-            - name: Install dependencies with composer
-              run: composer update --prefer-dist --no-interaction --no-progress --optimize-autoloader --ansi
-
-            - name: Install npm dependencies
-              run: npm install
-
-            - name: Build production assets
-              run: grunt build-assets
-
-            - name: Run migrations
-              run: php protected/humhub/tests/codeception/bin/yii migrate/up --includeModuleMigrations=1 --interactive=0
-
-            - name: Run installer
-              run: php protected/humhub/tests/codeception/bin/yii installer/auto
-
-            - name: Checkout REST Module
-              uses: actions/checkout@v2
-              with:
-                  repository: 'humhub/rest'
-                  ref: 'master'
-                  path: 'protected/modules/rest'
-
-            - name: Activate REST Module
-              run: php protected/humhub/tests/codeception/bin/yii module/enable rest
-
-            - name: Rebuild search index
-              run: php protected/humhub/tests/codeception/bin/yii search/rebuild
-
-            - name: Build codeception core files
-              run: |
-                cd $GITHUB_WORKSPACE/protected/humhub/tests
-                php ../../vendor/bin/codecept build
-
-            - name: Run test server
-              run: |
-                # Temporary condition while branch "master" doesn't use pretty URLs on acceptance tests
-                if cat $GITHUB_WORKSPACE/.github/workflows/php-test.yml | grep -q ":8080 index-test.php"
-                then
-                  php --server 127.0.0.1:8080 index-test.php &>/tmp/phpserver.log &
-                else
-                  php --server 127.0.0.1:8080 &>/tmp/phpserver.log &
-                fi
-
-            - name: Setup chromedriver
-              run: chromedriver --url-base=/wd/hub &
-
-            - name: Valdiate test server
-              run: sleep 5 && curl --fail --head http://127.0.0.1:8080/index-test.php
-
-            - name: Install module dependencies
-              run: |
-                cd $GITHUB_WORKSPACE/protected/modules/${{ env.module-id }}
-                composer update --prefer-dist --no-interaction --no-progress --optimize-autoloader --ansi
-
-            - name: Run test suite
-              run: |
-                export HUMHUB_PATH=$GITHUB_WORKSPACE
-                cd $GITHUB_WORKSPACE/protected/modules/${{ env.module-id }}/tests
-                php $GITHUB_WORKSPACE/protected/vendor/bin/codecept build
-                php $GITHUB_WORKSPACE/protected/vendor/bin/codecept run --env github
+  tests:
+    uses: humhub/actions/.github/workflows/module-tests.yml@main
+    with:
+      module-id: calendar
+      php: >-
```

Comment:
@luke- Yes, I agree to set a default value for the php version param, but we should to decide what default versions because some modules has min HumHub version for example 1.9 as here https://github.com/humhub/news/blob/main/module.json#L10, so this module can be tested only on php 7.3 - 8.0, but HumHub version 1.12 should be tested on php 7.4 - 8.1.

We can create several action files for each HumHub vesion like `module-tests-1.9.yml`, `module-tests-1.10.yml` and etc., so each file will call main `module-tests.yml` with param `php` for the requested HumHub version. So when module param `humhub:minVersion` will be increased then we just switch to the proper action test file as well. Do you agree this?

Reply:
@luke- 

> Can you please build the same for the Marketplace Update?
>
> Another point for the module tests, maybe we can add also a additonal schedule:

Ok, I will review these 2 points.

Reply:
> @luke- Yes, I agree to set a default value for the php version param, but we should to decide what default versions because some modules has min HumHub version for example 1.9 as here https://github.com/humhub/news/blob/main/module.json#L10, so this module can be tested only on php 7.3 - 8.0, but HumHub version 1.12 should be tested on php 7.4 - 8.1.
> 
> We can create several action files for each HumHub vesion like `module-tests-1.9.yml`, `module-tests-1.10.yml` and etc., so each file will call main `module-tests.yml` with param `php` for the requested HumHub version. So when module param `humhub:minVersion` will be increased then we just switch to the proper action test file as well. Do you agree this?

I like the idea! Can we create following files?

- `module-test-master.yml`  (With current `master` PHP requirements, against `master` branch only)
- `module-test-develop.yml` (With new `develop` PHP requirements, against `develop`)
- `module-test-v1.10.yml` (With new `v1.10` PHP requirements, against `v1.10` branch)
- ...

Each module should have at least `-master.yml` and `-develop.yml` tests. Ideally based on the `minVersion` in the `module.json` additional odler version `v1.10.yml` test files?

Important would be that all these test files simply include one central `module-test.yml`.